### PR TITLE
refactor: remove few unnecessary escapes from regex patterns

### PR DESCRIPTION
**Pull request by simPod**
**What is this feature?**

Remove few unnecessary escapes from regex patterns

**Why do we need this feature?**

Better readability

### CODE REVIEW

File: packages/grafana-data/src/utils/logs.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { ArrayVector } from '../vector/ArrayVector';
 // first a label from start of the string or first white space, then any word chars until "="
 // second either an empty quotes, or anything that starts with quote and ends with unescaped quote,
 // or any non whitespace chars that do not start with quote
-const LOGFMT_REGEXP = /(?:^|\s)([\w\(\)\[\]\{\}]+)=(""|(?:".*?[^\\]"|[^"\s]\S*))/;
+const LOGFMT_REGEXP = /(?:^|\s)([\w()\[\]{}]+)=(""|(?:".*?[^\\]"|[^"\s]\S*))/;
```
Comment (gabor) [inline packages/grafana-data/src/utils/logs.ts:11]: hi @simPod , thanks for the PR! regarding this change to `LOGFMT_REGEXP`, this is only called from functions marked as `obsolete`, they will be removed in the future, so right now i don't think we want to make changes to this part of the code.

(i cannot comment on the remaining changes in the PR)

thanks!

Comment (simPod) [reply packages/grafana-data/src/utils/logs.ts:11]:
```suggestion
const LOGFMT_REGEXP = /(?:^|\s)([\w\(\)\[\]\{\}]+)=(""|(?:".*?[^\\]"|[^"\s]\S*))/;
```

**Comment by leeoniya** [comment]
@simPod i'm not sure about this change, TBH. regexps are hard enough to parse as it is. i personally prefer to have all regexp special chars explicitly escaped without having to visually backtrack to discover that the opening brace was escaped so technically the closing brace dorsn't need it.

**Comment by simPod** [comment]
Aight, np

**Pull request closed without merging**